src/Share/Survey/SurveyHelper.cs

```diff
@@ -0,0 +1,360 @@
+        { 
+            return new ScheduleInfo() { LastPromptDate = LastPromptDate.ToString("yyyy-MM-dd"), Modules = Modules.Values.ToList() };
+        }
+
+        private bool ReadFromStream()
+        {
+            StreamReader sr = null;
+            try
+            {
+                if (File.Exists(SurveyScheduleInfoFile))
+                {
+                    sr = new StreamReader(new FileStream(SurveyScheduleInfoFile, FileMode.Open, FileAccess.Read, FileShare.None));                    
+                    MergeScheduleInfo(JsonConvert.DeserializeObject<ScheduleInfo>(sr.ReadToEnd()));
+                }
+            }
+            catch (Exception e)
+            {
+                if (e is UnauthorizedAccessException)
+                {
+                    this.IgnoreSchedule = true;
+                }
+                //deserialize failed, means content of file is incorrect, make file empty
+                if (e is JsonException)
+                {
+                    if (sr != null)
+                    {
+                        sr.Dispose();
+                    }
+                    EmptyFileAsync();
+                }
+                return false;
+            }
+            finally
+            {
+                if (sr != null)
+                {
+                    sr.Dispose();
+                }
+            }
+            return true;
+        }
+
+        private bool WriteToStream(string info)
+        {
+            StreamWriter sw = null;
+            try
+            {
+                sw = new StreamWriter(new FileStream(SurveyScheduleInfoFile, FileMode.Create, FileAccess.Write, FileShare.None));
+                sw.Write(info);
+            }
+            catch (Exception e)
+            {
+                if (e is UnauthorizedAccessException)
+                {
+                    this.IgnoreSchedule = true;
+                }
+                return false;
+            }
+            finally
+            {
+                if (sw != null)
+                {
+                    sw.Dispose();
+                }
+            }
+            return true;
+        }
+
+        private async void FlushAsync()
+        {
+            await Task.Run(() =>
+            {
+                WriteToStream(JsonConvert.SerializeObject(GetScheduleInfo()));
+            });
+        }
+
+        private async void EmptyFileAsync()
+        {
+            await Task.Run(() =>
+            {
+                WriteToStream(string.Empty);
+            });
+        }
+
+        private async void FlushWithWaitAsync()
+        {
+            await Task.Run(() =>
+            {
+                FlushWithWait();
+            });
+        }
+
+        private void FlushWithWait()
+        {
+            int beforeExchange = Interlocked.CompareExchange(ref FlushCount, 0, FlushFrequecy);
+            if (beforeExchange < FlushFrequecy)
+            {
+                Interlocked.Add(ref FlushCount, 1);
```

Comment:
`Interlocked.Increment` is doing "+1". It won't be worse than `Add` performance wise.

Reply:
done